### ISSUE TITLE
Rule suddenly scanning excluded directory

### ISSUE THREAD

**Issue by zeitchef** [tfeldmann/organize#339]
Before upgrading to v3, I had a rule that would scan my system for pesky `.DS_Store` files and delete them - excluding `~/Library`. Since upgrading and with no change to the rule, `~/Library` is being scanned as well which takes a lot of time.

- OS: macOS 14.2.1
- Output of `organize --version`: v3.0.0

Here is the relevant rule config:

```yaml
rules:
  # Delete .DS_Store files
  - name: 'Delete .DS_Store files'
    locations:
      - path: '~/'
        exclude_dirs: ['Library', 'Dropbox', '.Trash', 'Pictures']
        system_exclude_dirs: 
          ['.git', 'node_modules', '.docker', '.local', '.archive']
    subfolders: true
    filters:
      - name: .DS_Store
    actions:
      - delete
```

**Comment by tfeldmann** [comment]
Thank you for reporting this issue! This should be fixed in f9f18726fa087be37f70b6cdf8fdc257efe2bf19. I'll release the new version as soon as possible.

**Comment by tfeldmann** [comment]
The fix is released!
